src/vs/editor/contrib/hover/browser/contentHover.ts

```diff
@@ -437,252 +442,385 @@ class ContentHoverVisibleData {
 	) { }
 }
 
-export class ContentHoverWidget extends Disposable implements IContentWidget {
+const HORIZONTAL_SCROLLING_BY = 30;
+const SCROLLBAR_WIDTH = 10;
+const SASH_WIDTH_MINUS_BORDER = 3;
 
-	static readonly ID = 'editor.contrib.contentHoverWidget';
+export class ResizableHoverWidget extends ResizableContentWidget {
 
-	public readonly allowEditorOverflow = true;
+	public static ID = 'editor.contrib.resizableContentHoverWidget';
 
-	private readonly _hoverVisibleKey = EditorContextKeys.hoverVisible.bindTo(this._contextKeyService);
-	private readonly _hoverFocusedKey = EditorContextKeys.hoverFocused.bindTo(this._contextKeyService);
-	private readonly _hover: HoverWidget = this._register(new HoverWidget());
-	private readonly _focusTracker = this._register(dom.trackFocus(this.getDomNode()));
-	private readonly _horizontalScrollingBy: number = 30;
-	private _visibleData: ContentHoverVisibleData | null = null;
+	private _disposableStore = new DisposableStore();
```

Comment:
Bug: This isn't actually disposed. Also you inhert a store (and _register-function) from the super class that could be using